content/4. Deeltaken/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/While/1. While.md

```diff
@@ -0,0 +1,352 @@
+---
+title: 1. While
+taxonomie:
+  - ib-19.2.Controlestructuur-Herhaling-While.DT
+  - ib-19.3.Controlestructuur-Herhaling-While.DT
+---
+
+> Meer informatie over [[content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/While/1. While|While]]
+
+## Opdracht 1
+Zet de volgende casus om naar een **While**-loop.
+
+### Specificaties
+* Er wordt een random getal gekozen.
+* Het programma blijft vragen om een invoer tot het juiste getal is geraden (bijvoorbeeld 7).
+* Ongeacht de uitkomst, print de uitkomst in de console.
+
+### Verwachte output:
+Als `gok != juisteGetal`, dan komt in de console:
+```
+Raad het getal (tussen 1 en 10):
+Helaas dat is niet correct.
+Raad het getal (tussen 1 en 10):
+```
+Als `gok == juisteGetal`, dan komt in de console::
+```
+Gefeliciteerd! Dat is correct.
+```
+
+### Nu jij:
+``` csharp runner
+using System;
+
+// Het juiste getal  
+int correctNumber = 7;  
+  
+// Kies een random getal.  
+Random rand = new Random();  
+int number = rand.Next(5, 8);  
+  
+// While-lus om te blijven raden  
+while ()  
+{  
+    //Kies een nieuw random getal  
+    number = rand.Next(5, 8);  
```

Comment:
Haha, okeee die is wel een béétje flauw! (staat namelijk niet in de specificatie dat -binnen de loop- steeds een nieuw getal wordt gekozen)
Dus wanneer je 7 raad (en dit incorrect is), kan je in principe 7 blijven zeggen totdat je het een keer goed hebt?
En andersom alle getallen van 1 t/m 10 opnoemen zónder het één keer goed te hebben?

Edit: Oké, denk dat ik 'm door heb. Er wordt wel om invoer gevraagd, maar eigenlijk wordt deze gegeven/randomly bepaald?

Reply:
Ah, dit is inderdaad een foutje van mij. Eerder wilde ik deze opgave(s) maken met een Console.ReadLine(), de coderunner ondersteunt deze functie niet dus heb alles herschreven naar random getallen.

Omdat het gebruik van Random buiten de scope van While ligt heb ik dit niet benoemd en bij de opdracht meegeven, alleen nooit aangepast in de specificaties, oeps.. Dit is bij de volgende push gefixt :)